src/freenet/node/useralerts/UserAlertManager.java

```diff
@@ -158,6 +158,13 @@ public int compare(UserAlert a0, UserAlert a1) {
 			int classHash1 = a1.getClass().hashCode();
 			if(classHash0 > classHash1) return 1;
 			else if(classHash0 < classHash1) return -1;
+			// then go by sent time, if both are n2ntms
+			boolean isN2NTM0 = a0 instanceof N2NTMUserAlert;
+			boolean isN2NTM1 = a1 instanceof N2NTMUserAlert;
+			if(isN2NTM0 && isN2NTM1) { // newest first
+				if(a0.getUpdatedTime() < a1.getUpdatedTime()) return 1;
+				else if(a0.getUpdatedTime() > a1.getUpdatedTime()) return -1;
+			}
```

Comment:
Ordering by updated time is always more sensible than ordering by hash code. I don't see the need to restrict this to `N2NTMUserAlert`. Apart from that, I'd prefer the `UserAlertManager` to be as agnostic as possible of the kind of `UserAlert`.